application/common/mail/refresh-backup-codes.html.php

```diff
@@ -22,8 +22,34 @@
  * @var string $supportEmail
  * @var string $supportName
  * @var bool   $isMfaEnabled
+ * @var int    $numRemainingCodes
  */
 ?>
 <p>Dear <?= yHtml::encode($displayName) ?>,</p>
 
-<h2>TODO: Compose contents of this email message</h2>
\ No newline at end of file
+<p>
+    You only have <?= $numRemainingCodes ?> Printable Codes remaining for use with 2-Step Verification on your
```

Comment:
Do we want to treat this any differently for 1 (or even for 0) codes remaining? We could even keep it as simple as using `Code(s)` as you did elsewhere.

Reply:
Done